src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,162 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+
 namespace System.Globalization
 {
-    using System;
-    using System.Text;
-    using System.Collections;
-    using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Versioning;
-    using System.Security;
-    using System.Threading;
-    using System.Diagnostics.Contracts;
-
     internal static class EncodingTable
     {
+        //
+        // This caches the data item which is indexed by the code page value.
```

Comment:
I ran some quick perf measurements (10,000,000 iteration micro-benchmark) on full framework 4.6 on Windows 10 x64 (ryujit)...

 * Using a pre-allocated static `Dictionary<int, CodePageDataItem>` is ~2x faster than the current lookup with `Hashtable`.
 * Using a `switch` is ~9.3x faster than `Hashtable`, and ~5.5x faster than `Dictionary`.

```c#
internal static CodePageDataItem GetCodePageDataItem(int codepage)
{
    switch (codepage)
    {
        case 1200: return encodingDataTableItems[0];
        case 1201: return encodingDataTableItems[1];
        case 12000: return encodingDataTableItems[2];
        case 12001: return encodingDataTableItems[3];
        case 20127: return encodingDataTableItems[4];
        case 28591: return encodingDataTableItems[5];
        case 65000: return encodingDataTableItems[6];
        case 65001: return encodingDataTableItems[7];
    }

    return null;
}
```